Extract MD5 logic for reuse

Kangie opened the pull request:
Gentoo's portage also uses MD5s to track which files have changed since installation.

Added some debug statements for later.

Kangie commented:
I'm going to have to rely on CI/CD to make sure that I didn't break anything on *buntu, I can't get Gentoo's `dpkg` package to spit out a libdpkg that we can link against. Changes are pretty innocuous IMO

stevegrubb commented:
OK, looking at the change to configure.ac, it adds AC_USE_SYSTEM_EXTENSIONS which was introduced in autoconf-2.60. But we are declaring conformance to 2.50. As far as I know, fapolicy only runs on Linux. So, we can add _GNU_SOURCE on a case by case basis. until there's a need to rebase to a newer autoconf. Looks like RHEL 8 has autoconf-2.69 available. I don't know about other Linux distributions. Still reviewing...

Kangie commented:
Looks like RHEL 8 has autoconf-2.69 available.

2.60 came out in 2006, so I think we're fine here; I can't imagine a reasonable system running a kernel that supports FANOTIFY and not having 2.60. I will update that to target 2.60 though if you're happy with that?

> ... But if you do that, maybe it's better not to separate the code in test_file() from add_file_to_backend_by_md5() ?

I'll rework that, thanks.

stevegrubb commented:
Thanks. Looks good to me.

Kangie commented:
I'm happy if you are. I haven't addressed your feedback about putting the uthash stuff in each C file that calls it; ran out of time before work - If that's fine, great, if not I'm equally happy to make those changes :smile:
